lec02-basic-r.Rmd

```diff
@@ -1,4 +1,754 @@
+workhorse of R. A vector is composed by a series of values, which can be either
+numbers or characters. We can assign a series of values to a vector using the
+`c()` function, which stands for "concatenate (combine/connect one after
+another) values into a vector" For example we can create a vector of animal
+weights and assign it to a new object `weight_g`:
+
+```{r}
+weight_g <- c(50, 60, 65, 82) # Concatenate/Combine values into a vector
+weight_g
+```
+
+You can also use the built-in command `seq`, to create a sequence of numbers
+without typing all of them in manually.
+
+```{r}
+seq(0, 30)
+seq(0, 30, 3) # Every third number
+```
+
+A vector can also contain characters:
+
+```{r}
+animals <- c('mouse', 'rat', 'dog')
+animals
+```
+
+The quotes around "mouse", "rat", etc. are essential here and can be either
+single or double quotes. Without the quotes R will assume there are objects
+called `mouse`, `rat` and `dog`. As these objects don't exist in R's memory,
+there will be an error message.
+
+There are many functions that allow you to inspect the content of a
+vector. `length()` tells you how many elements are in a particular vector:
+
+```{r}
+length(weight_g)
+length(animals)
+```
+
+An important feature of a vector, is that all of the elements are the same type
+of data. The function `class()` indicates the class (the type of element) of an
+object:
+
+```{r}
+class(weight_g)
+class(animals)
+```
+
+The function `str()` provides an overview of the structure of an object and its
+elements. It is a useful function when working with large and complex
+objects:
+
+```{r}
+str(weight_g)
+str(animals)
+```
+
+You can use the `c()` function to add other elements to your vector:
+
+```{r}
+weight_g <- c(weight_g, 90) # add to the end of the vector
+weight_g <- c(30, weight_g) # add to the beginning of the vector
+weight_g
+```
+
+In the first line, we take the original vector `weight_g`,
+add the value `90` to the end of it, and save the result back into
+`weight_g`. Then we add the value `30` to the beginning, again saving the result
+back into `weight_g`.
+
+We can do this over and over again to grow a vector, or assemble a dataset.
+As we program, this may be useful to add results that we are collecting or
+calculating.
+
+An **atomic vector** is the simplest R **data type** and it is a linear vector
+of a single type, e.g. all numbers. Above, we saw 2 of the 6 main **atomic
+vector** types  that R uses: `"character"` and `"numeric"` (or `"double"`).
+These are the basic building blocks that all R objects are built from. The other
+4 **atomic vector** types are:
+
+* `"logical"` for `TRUE` and `FALSE` (the boolean data type)
+* `"integer"` for integer numbers (e.g., `2L`, the `L` indicates to R that it's an integer)
+* `"complex"` to represent complex numbers with real and imaginary parts (e.g.,
+  `1 + 4i`) and that's all we're going to say about them
+* `"raw"` for bitstreams that we won't discuss further
```

Comment:
Not sure if I should even mention this or just introduce logicals, since they will use them for conditional subsetting. Same for the challenge just below.

Reply:
i think it's fine for now until you eventually introduce logicals...?